src/script/descriptor.h

```diff
@@ -108,6 +108,9 @@ struct Descriptor {
     /** Convert the descriptor back to a string, undoing parsing. */
     virtual std::string ToString() const = 0;
 
+    /* Constant descriptor ID. */
+    virtual std::string GetID() const = 0;
```

Comment:
In 494fa79a421c7384fdecacc7b42ae0dbe66668f2 "wallet: bugfix, always use apostrophe for spkm descriptor ID"

This name is misleading as it doesn't actually return an id, just the string we hash for it. I would actually prefer just adding a parameter to `ToString()` rather than a whole new function.

Reply:
I did that first, but didn't like to add a `force_apostrophe` argument to it. The apostrophe existence is internal to the bip32 `PubKeyProvider` only and not really a concept that the `Descriptor` class should know about. 
Even other pub key providers shouldn't be aware of the apostrophe format neither.

The idea behind the `GeID()` method was to have a function that we treat differently and think twice before change anything inside it.

Said that, I'm not that strong over it neither. Could use the same `ID` concept for the `ToString` parameter too.